identity/credentials.go

```diff
@@ -248,7 +249,13 @@ func CredentialsEqual(a, b map[CredentialsType]Credentials) bool {
 			return false
 		}
 
-		if string(expect.Config) != string(actual.Config) {
+		// Try to normalize configs (remove spaces etc).
+		patch, err := jsondiff.CompareJSON(actual.Config, expect.Config)
```

Comment:
This is a comprehensive comparison, ensuring that key shuffling or other things (e.g. extra spaces) don't cause trouble!